File: 2478/2478.js
/**
 * @param {TreeNode} root
 * @return {number}
 */
var numberOfBeautifulPartitions = function(root) {
    if (!root) return 0;
    return 1 + Math.max(numberOfBeautifulPartitions(root.left), numberOfBeautifulPartitions(root.right));
};

File: 2478/2478.go
func numberOfBeautifulPartitions(root *TreeNode) int {
	if root == nil {
		return 0
	}
	l := numberOfBeautifulPartitions(root.Left)
	r := numberOfBeautifulPartitions(root.Right)
	if l > r {
		return 1 + l
	}
	return 1 + r
}